addons/crm/models/crm_lead.py

```diff
@@ -495,7 +437,7 @@ def _rebuild_pls_frequency_table_threshold(self):
 
     def action_set_lost(self, **additional_values):
         """ Lost semantic: probability = 0 or active = False """
-        result = self.write({'active': False, 'probability': 0, **additional_values})
+        result = self.write({'active': False, 'probability': 0, 'automated_probability': 0, **additional_values})
```

Comment:
Good because we won't get an estimated probability = x% when lost.
But when you restore, as the automated probability is not recomputed, the side effect of this is that the probability of the lead will still be 0 but also the estimated probability. So it looks like PLS is not working anymore when you restore.

I think we should force the recompute when restoring. (and as probability and automated probability are aligned, the lead will be restored in auto mode. which I think is good. - otherwise the probability will remains at 0% -)

Reply:
Is it ok with my last push?

Reply:
Keep using toggle_active and just override the method in crm_lead model.
(toggle_active is defined in odoo/models.py)